azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -158,3 +158,40 @@ def test_proxy_token_providers(self, mock_get, proxy_kcsb):
+            "https://kusto.test.com:4242": "https://kusto.test.com:4242",
+            "https://kusto.test.com:4242/": "https://kusto.test.com:4242",
+            "https://kusto.test.com:4242/test": "https://kusto.test.com:4242/test",
+            "https://kusto.test.com;fed=true": "https://kusto.test.com",
+            "https://kusto.test.com/;fed=true": "https://kusto.test.com",
+            "https://kusto.test.com/test;fed=true": "https://kusto.test.com/test",
+            "https://kusto.test.com:4242;fed=true": "https://kusto.test.com:4242",
+            "https://kusto.test.com:4242/;fed=true": "https://kusto.test.com:4242",
+            "https://kusto.test.com:4242/test;fed=true": "https://kusto.test.com:4242/test",
+            "https://ade.loganalytics.io/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace": "https://ade.loganalytics.io/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace",
+            "https://ade.loganalytics.io/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace/": "https://ade.loganalytics.io/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace",
+            "https://ade.loganalytics.io:4242/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace/": "https://ade.loganalytics.io:4242/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace",
+            "https://ade.loganalytics.io:4242/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65/resourcegroups"
+            "/some_resource_group/providers/microsoft.operationalinsights/workspaces/some_workspace/;fed=true": "https://ade.loganalytics.io:4242/subscriptions/da45f7ac-97c0-4fff-ac66-3b6810eb4f65"
+            "/resourcegroups/some_resource_group/providers/microsoft.operationalinsights/workspaces"
+            "/some_workspace",
+            "https://kusto.aria.microsoft.com": "https://kusto.aria.microsoft.com",
+            "https://kusto.aria.microsoft.com/": "https://kusto.aria.microsoft.com",
+            "https://kusto.aria.microsoft.com/;fed=true": "https://kusto.aria.microsoft.com",
+        }
+        for entry_key, entry_value in tests.items():
+            kcsb = KustoConnectionStringBuilder.with_aad_user_token_authentication(entry_key, "test")
+            client = KustoClient(kcsb)
+            assert (entry_value, client._kusto_cluster)
```

Comment:
Sadly that's the wrong form of assert, this will always return true.

You want to do 
`assert entry_value == client._kusto_clutser`

And we already see some of the tests fail